docker2earth/convert.go

```diff
@@ -0,0 +1,116 @@
+package docker2earth
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/earthly/earthly/fileutils"
+
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
+	"github.com/moby/buildkit/frontend/dockerfile/parser"
+	"github.com/pkg/errors"
+)
+
+func getArtifactName(s string) string {
+	split := strings.Split(s, "/")
+	n := len(split)
+	return split[n-1]
+}
+
+// Docker2Earth converts an existing Dockerfile in the current directory and writes out an Earthfile in the current directory
+// and error is returned if an Earthfile already exists.
+func Docker2Earth() error {
+	if fileutils.FileExists("Earthfile") {
+		return fmt.Errorf("Earthfile already exists; please delete it if you wish to continue")
+	}
+
+	in, err := os.Open("Dockerfile")
```

Comment:
It might be nice to have this work via stdin & stdout, so it can be piped like typical unix commands. This could be a future thing though.

Reply:
just opened up a PR for this: https://github.com/earthly/earthly/pull/474